### DISCUSSION TITLE
List files from Index Of HTML pages

### DISCUSSION THREAD

**Discussion by dufferzafar** [saulpw/visidata#1424] [Ideas]
Currently, index of pages like: https://hls.gsfc.nasa.gov/data/ are parsed as a table

It would be great if each row could be a "link" to the file - which could then be opened in visidata itself.

Will help with html sites that contain a lot of CSVs. This would be quite similar to what happens when you run visidata on a directory.

**Comment by saulpw** [comment]
Thanks for the idea, @dufferzafar!  I've added a simple links sheet to the html loader, which will show every link on the page (likely including many that you don't care about).  You can also press Enter on one of those links to open it in VisiData. This is a first cut and I'm sure there's more that can be done, but hopefully this helps make it easier to get data from web pages.

**Comment by frosencrantz** [comment]
Thanks @saulpw!  And thanks for the suggestion @dufferzafar!

I like this new sheet of links.

There might be something wrong with the way links are found in tables, For example, with the link from the original request, there is a stack trace for the added link columns in the table sheets.

https://asciinema.org/a/oovSB1lpjNyXUYm3T7N7DAmHt

**Reply by frosencrantz** [reply]
One feature I would want, is the ability to jump directly to the table of all lines when diving into a link.  As written it dives into a HTMLTablesSheet (which lists all the tables & the links sheet) and it would be nice if there was a way to also dive directly to the HTMLLinksSheet.  Even if I dived directly into the HTMLLinksSheet, there should still be a way to get to the related HTMLTablesSheet for the current links sheet.
